docs/args-and-options.md

```diff
@@ -0,0 +1,180 @@
+Command(name, optionsNone, argsNone)
+```
+
+## Args
+`Args[A]` is a description of the process of constructing an instance of `A` from a valid input of the CLI. It is not yet a specified argument for the CLI. In other words, an `Args[A]` defines a collection of valid arguments for a command and a way to construct a value `A` from them.
+
+**ZIO CLI** offers a variety of methods to create basic `Args` that take a string from the user as input and transform it into a value of the corresponding type. 
+```scala mdoc:silent
+// Basic constructions
+Args.bool(name)
+Args.file(name)
+Args.directory(name)
+Args.text(name)
+Args.decimal(name)
+Args.integer(name)
+Args.instant(name)
+Args.localDate(name)
+Args.localDateTime(name)
+Args.localTime(name)
+Args.monthDay(name)
+Args.offsetDateTime(name)
+Args.offsetTime(name)
+Args.period(name)
+Args.year(name)
+Args.yearMonth(name)
+Args.zonedDateTime(name)
+Args.zoneId(name)
+Args.zoneOffset(name)
+```
+We can also avoid specifying a name of the `Args`. In this case, the name will be the type:
+```scala mdoc:silent
+Args.bool(name) // Boolean Args with name
+Args.bool       // Boolean Args named "boolean"
+```
+
+### Combining and transforming Args
+It is possible to create more complex `Args` using the following methods of the trait `Args`:
+```scala mdoc:silent
+trait Args[A] {
+  def ++[B](that: Args[B]): Args[(A, B)]              // Zip two args
+  def +[A1 >: A]: Args[::[A1]]                        // Requires a non-empty list of Args of type A1
+  def * : Args[List[A]]                               // Requires a list of Args of same type
+  def atLeast(min: Int): Args[List[A]]                // Requires a list of Args with at least min elements
+  def atMost(max: Int): Args[List[A]]                 // Requires a list of Args with at most max elements
+  def between(min: Int, max: Int): Args[List[A]]      // Requires a list of Args with a bound on the number of elements
+  def map[B](f: A => B): Args[B]                      // Applies a function f to the result of the Args
+  def ??(that: String): Args[A]                       // Adds a string to the HelpDoc of the Args
+}
+```
+
+### Example
+We are going to construct an `Args` that asks for a list of 12 decimals (one for each month) and a year. Then it creates a new `Args` that stored the year and the mean value.
+```scala mdoc:silent:reset
+import zio.cli._
+
+val data: Args[BigDecimal] = Args.decimal
+val args: Args[(List[BigDecimal], BigInt)] = data.between(12, 12) ++ Args.integer
+
+case class YearAndMean(year: BigInt, mean: BigDecimal)
+
+val mappedArgs: Args[YearAndMean] = args.map {
+  case (months, year) => YearAndMean(year, months.sum/12)
+}
+```
+
+## Options
+`Options[A]` is a description of the process of constructing an instance of `A` from a valid input of the CLI. It is not yet a specified option for the CLI. In other words, an `Options[A]` defines a collection of valid commands and a way to construct a value `A` from them.
```

Comment:
Same for introducing `Options`: we need to precisely describe what options are, how they differ from arguments, and give multiple examples across command-line tools that a developer is probably familiar with already.